python/tskit/tables.py

```diff
@@ -2146,14 +2180,58 @@ def __str__(self):
         s += str(self.provenances)
         return s
 
-    def __eq__(self, other):
+    def equals(
+        self,
+        other,
+        ignore_metadata=False,
+        ignore_ts_metadata=False,
+        ignore_provenance=False,
+        ignore_timestamps=False,
+    ):
+        """
+        Returns True if  `self` and `other` are equal. By default, two table
+        collections are considered equal if their
+
+        - ``sequence_length`` properties are identical;
+        - top-level tree sequence metadata and metadata schemas are
+          byte-wise identical;
+        - constituent tables are byte-wise identical.
+
+        Some of the requirements in this definition can be relaxed using the
+        parameters, which can be used to remove certain parts of the data model
+        from the comparison.
+
+        Table indexes are not considered in the equality comparison.
+
+        :param TableCollection other: Another table collection.
+        :param bool ignore_metadata: If True *all* metadata and metadata schemas
+            will be excluded from the comparison. This includes the top-level
+            tree sequence and constituent table metadata (default=False).
+        :param bool ignore_ts_metadata: If True the top-level tree sequence
+            metadata and metadata schemas will be excluded from the comparison.
+            If ``ignore_metadata`` is True, this parameter has no effect.
+        :param bool ignore_provenance: If True the provenance tables are
+            not included in the comparison.
+        :param bool ignore_timestamps: If True the provenance timestamp column
+            is ignored in the comparision.
```

Comment:
Add a note that there is no effect if `ignore_provenance` it is there in the c doc and for `ignore_ts_metadata`

Reply:
Good point, well spotted.